backend/migrations/dataall_migrations/migrationmanager.py

```diff
@@ -0,0 +1,124 @@
+            self.current_migration = self.current_migration.next_migration()
+            if not self.current_migration:
+                raise Exception(f'Revision {current_revision} is not found.')
+
+    def save_upgraded(self, executed_ups: Deque[Union[Type[MigrationBase], None]]):
+        while executed_ups:
+            self.previous_migrations.append(executed_ups.popleft())
+
+    def save_downgrades(self, executed_downs: Deque[Union[Type[MigrationBase], None]]):
+        while executed_downs:
+            try:
+                self.previous_migrations.remove(executed_downs.pop())
+            except Exception as e:
+                ...
+
+    def check_downgrade_id(self, target_revision_id):
+        for pm in self.previous_migrations:
+            if pm.revision_id() == target_revision_id:
+                return True
+        if target_revision_id == self.current_migration.revision_id():
+            return False
+
+        raise Exception(f'Failed to find {target_revision_id} in migration history.')
+
+    def check_upgrade_id(self, target_revision_id):
+        if target_revision_id is None:
+            if (
+                self.current_migration.next_migration() is None
+                or self.current_migration.revision_id() == target_revision_id
+            ):
+                return False
+            return True
+
+        revision = self.current_migration.next_migration()
+        while revision is not None:
+            if revision.revision_id() == target_revision_id:
+                return True
+            revision = revision.next_migration()
+
+        Exception(f'Failed to find {target_revision_id}.')
+
+    def upgrade(self, target_revision_id=None):
+        if not self.check_upgrade_id(target_revision_id):
+            logger.info('Data-all version is up to date')
+            return self.current_migration.revision_id()
+
+        logger.info(f"Upgrade from {self.current_migration.revision_id()} to {target_revision_id or 'latest'}")
+        executed_upgrades: Deque[Union[Type[MigrationBase], None]] = deque()
+        saved_start_migration = self.current_migration
+        while self.current_migration is not None:
+            try:
+                logger.info(f'Applying migration {self.current_migration.__name__}')
+                self.current_migration.up()
+                executed_upgrades.append(self.current_migration)
+                logger.info(f'Migration {self.current_migration.__name__} completed')
+                if (
+                    self.current_migration.revision_id() == target_revision_id
+                    or self.current_migration.next_migration() is None
+                ):
+                    break
+                self.current_migration = self.current_migration.next_migration()
+            except Exception as e:
+                # Something went wrong revert
+                logger.info(f'An error occurred while applying the migration.{e}.')
```

Comment:
This will only print the exception message, I think it would be nice if we can see the stack trace here. Try `logger.exception('An error occurred while applying the migration.')` [docs](https://docs.python.org/3/library/logging.html#logging.Logger.exception) which does that by default.